### ISSUE TITLE
Psalm incorrectly warns about `continue;` in switch statement

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#464]
Inside of a `switch` statement, `continue` is an alias of `break`. 

- `continue <number_of_levels:int>;` is possible, but I'm not aware of type checkers that go to the lengths of analyzing that. (Might make sense to warn)

Noticed when looking at SwitchChecker.php.

This is valid code, and can be executed outside of a loop:

Observed:  ContinueOutsideLoop - ... - Continue called when not in loop
Expected: Should not warn, at least for that reason
```php
<?php

switch(2) {
    case 2:
        echo "two\n";
        continue;
}
```

The below code is also valid, and will print both "In switch" and "After switch"

```
foreach ([2] as $x) {
    switch($x) {
        default:
        echo "In loop x=$x\n";
        continue;
    }
    echo "After switch x=$x\n";
}
```

---

https://secure.php.net/manual/en/control-structures.continue.php

>     Note: In PHP the switch statement is considered a looping structure for the purposes of continue. continue behaves like break (when no arguments are passed). If a switch is inside a loop, continue 2 will continue with the next iteration of the outer loop.

**Comment by muglug** [comment]
> Inside of a `switch` statement, `continue` is an alias of `break`.

(╯°□°）╯︵ ┻━┻

**Comment by TysonAndre** [comment]
Emitting ConfusingContinueInSwitch and falling through (not returning false) would probably make the most sense, due to the amount of confusion developers have about what it actually does.

**Comment by muglug** [comment]
I'm opting for the path of least resistance – treating `continue` as `break`, unless it's `continue 2` in which case it gets treated as `continue`, if we're inside a `switch`.
